### ISSUE TITLE
Empty BOM file is created for project using Arcade tooling

### ISSUE THREAD

**Issue by Bertk** [CycloneDX/cyclonedx-dotnet#210]
Hi,
I use [Microsoft dotnet Arcade tools](https://github.com/dotnet/arcade) for csharp projects and CyclonDX does always create a empty BOM file whenever the Arcade extensions are activated. After removing the Arcade tooling (global.json, Directory.Build.props, Directory.Build.targets) a complete BOM is created.  

How to reproduce: clone Arcade repository and run CycloneDX tool: `dotnet CycloneDX Arcade.sln --out lincenseDX`

The generated BOM.xml file:
```xml
<?xml version="1.0" encoding="utf-8"?>
<bom version="1" serialNumber="urn:uuid:c4bc047c-9409-4f19-aa68-b7aad0a195f5" xmlns="http://cyclonedx.org/schema/bom/1.1">
  <components />
</bom>
```

**Comment by coderpatros** [comment]
Hi @Bertk, thanks for the report and repro.

**Comment by Bertk** [comment]
I checked the sources and found a statement which will not work whenever the MSBuild output folder is not a subfolder of the csharp projects folder.

**_cyclonedx-dotnet/CycloneDX.Core/Services/ProjectFileService.cs_**
```            if (restoreResult.Success)
            {
                var assetsFilename = _fileSystem.Path.Combine(
                    _fileSystem.Path.GetDirectoryName(projectFilePath),
                    "obj", "project.assets.json");
                
                packages.UnionWith(_projectAssetsFileService.GetNugetPackages(assetsFilename));
            }
```

The folder structure with Arcade is different: **artifacts/obj/$(MSBuildProjectName)/$(Configuration)**
https://github.com/dotnet/arcade/blob/master/Documentation/ArcadeSdk.md#single-build-output

**Comment by Bertk** [comment]
@coderpatros I created a workaround which does not require additional dependencies.
Should I create a pull request using this [branch](https://github.com/Bertk/cyclonedx-dotnet/tree/fix-IntermediateOutputPath) or submit another branch with minimal updates.

**Comment by coderpatros** [comment]
@Bertk thank you for your PR. This is now available in v1.1.0
